awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -265,7 +265,14 @@ def get_one(self, endpoint, *args, **kwargs):
 
     def resolve_name_to_id(self, endpoint, name_or_id):
         # Try to resolve the object by name
-        response = self.get_endpoint(endpoint, **{'data': {'name': name_or_id}})
+        name_field = 'name'
+        if endpoint == 'users':
+            name_field = 'username'
+
+        response = self.get_endpoint(endpoint, **{'data': {name_field: name_or_id}})
+        if response['status_code'] == 400:
+            self.fail_json(msg="Unable to try and resolve {0} for {1} : {2}".format(endpoint, name_or_id, response['json']['detail']))
```

Comment:
I would favor raising `RuntimeError` for these cases. Because there's no _user_ input that should ever cause this. If a 400 error happens, it's because this method or some module was coded wrong. For coding errors, human-friendly messages aren't helpful, and if someone files a bug for it, a full traceback is much better, preferably with the full response data with it.